src/org/rascalmpl/library/lang/java/m3/AST.rsc

```diff
@@ -199,7 +361,7 @@ set[loc] findRoots(set[loc] folders) {
 @description{
 Wrapper around ((createAstsFromFiles)) to call it on a single file.
 }
-public Declaration createAstFromFile(loc file, bool collectBindings, bool errorRecovery = false, list[loc] sourcePath = [], list[loc] classPath = [], str javaVersion = "1.7") {
+public Declaration createAstFromFile(loc file, bool collectBindings, bool errorRecovery = false, list[loc] sourcePath = [], list[loc] classPath = [], Language javaVersion = JLS13()) {
```

Comment:
I like this enum :+1: but why default to java 13? I would expect one of the LTS versions like 8/11/17/21. And what happens if in the future we also add support for newer versions, will the default change?

(same for the others in this file)

Reply:
To make sure the porting to newer versions is always as small as possible, I would advice users to always go with the highest support we currently have, if possible. That way when we release support for newer versions, upgrading will always be between the lowest number of versions.

So yes, the default will always be the highest we have.

If there are other considerations, I'd be happy  to listen.

Reply:
I'm thinking of backwards compatibility, if I have code that's not setting this language level (because of the default, I'm not forced to commit to a specific version). Hypothetical say  we release this as 11. And I'm analyzing code with it that has some variables named 'yield' in there. Now we do some new work, and add Java 17 supported, and we switch the default to java 17. For that new version of rascal, my analysis failed, without me changing my code. (Since yield is now a keyword)

So either we freeze the default, or we make it an required parameter, and have a special 'highestSupported' kind of ADT member, for people that just want the lastest and greatest.

Reply:
I understand the argument but I don't agree setting an old version and fix it is a good solution for most cases. It is for the _yield_ example; but it would mean we now have to lock on Java 8, as this was the version we have supported up until now (and partially at that). Most students would have to set the version parameter to 12 or 13.

* if people need backwards compatibility for their actual compiler, they must set the version level explicitly, otherwise all compilers always default to their current/latest version. So to get this 100% right, we probably should add some code to the Maven support to extract that configuration parameter from the XML. But otherwise I think it is safe enough to follow the normal Java compiler behavior.
* `highestSupported` constructor is not right since M3 models need to be self-descriptive. The goal of the version parameter is not only to configure the compiler the right way, but also to log the chosen version as it is. The `Language` ADT is a specific ADT to encode language names and their versions from `analysis::m3::Core`.

Reply:
How do you feel about making it a positional argument?

And my 'latestVersion' would just be meant as a parameter, not as a valid value in an m3 model.